lua/luasnip/nodes/snippet.lua

```diff
@@ -21,6 +21,19 @@ local true_func = function()
 	return true
 end
 
+local generate_condition_only_matcher_func = function(condition)
+	return function(self, line_to_cursor, match, captures)
+		if condition(line_to_cursor, match, captures) then
+			return {
+				trigger = match,
+				captures = captures,
+			}
+		else
+			return nil
+		end
+	end
```

Comment:
Oh it should also process a user-provided context_matcher if the condition is true

Reply:
Ah, sorry. I misunderstand it. Fix it!